astronomer/providers/databricks/operators/databricks.py

```diff
@@ -1,17 +1,50 @@
+from __future__ import annotations
+
 from typing import Any
 
 from airflow.exceptions import AirflowException
+from airflow.providers.databricks.hooks.databricks import DatabricksHook
 from airflow.providers.databricks.operators.databricks import (
     XCOM_RUN_ID_KEY,
     XCOM_RUN_PAGE_URL_KEY,
     DatabricksRunNowOperator,
     DatabricksSubmitRunOperator,
+    RunState,
 )
 
 from astronomer.providers.databricks.triggers.databricks import DatabricksTrigger
 from astronomer.providers.utils.typing_compat import Context
 
 
+def _handle_non_successful_teminal_states(
+    run_state: RunState, run_info: dict[str, Any], hook: DatabricksHook, task_id: str
+) -> None:
+    """Raise AirflowException with detailed error message from run_info"""
```

Comment:
Same suggestion as in https://github.com/astronomer/astronomer-providers/pull/1084#discussion_r1193499247